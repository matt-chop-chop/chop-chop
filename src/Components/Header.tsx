import { Flex } from "@chakra-ui/react";
import { Logo } from "@/Components";

export const Header = () => {
  return (
    <Flex
      alignItems="center"
      background="var(--chakra-colors-light-surface)"
      borderBottom="1px solid var(--chakra-colors-light-emphasis)"
      position="fixed"
      px={4}
      py={2}
      width="100%"
      zIndex={1000}
    >
      <Logo />
    </Flex>
  );
};

export default Header;
