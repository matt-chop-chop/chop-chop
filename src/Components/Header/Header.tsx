import { Container, Flex, Link, Text } from "@chakra-ui/react";
import { Logo } from "@/Components";
import NextLink from "next/link";
import { useRouter } from "next/router";
import React from "react";

export const Header = () => {
  const router = useRouter();
  const { pathname } = router;

  const isHomeActive = pathname === "/";

  return (
    <Flex
      alignItems="center"
      background="var(--chakra-colors-light-emphasis)"
      borderBottom="1px solid var(--chakra-colors-light-text)"
      position="fixed"
      px={4}
      py={2}
      width="100%"
      zIndex={1000}
    >
      <Container maxWidth="1320px" width="100%">
        <Flex alignItems="center">
          <Logo />
          {!isHomeActive && (
            <Link
              as={NextLink}
              href="/"
              ml={6}
              transition="transform 250ms ease-out"
              _hover={{
                transition: "transform 250ms ease-in",
                transform: "scale(1.1)",
              }}
            >
              <Text variant="body-large">Home</Text>
            </Link>
          )}
        </Flex>
      </Container>
    </Flex>
  );
};

export default Header;
