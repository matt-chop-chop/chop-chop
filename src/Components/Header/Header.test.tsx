import { shallow } from "enzyme";
import React from "react";
import Header from "./Header";

jest.mock("next/router", () => ({
  __esModule: true,
  useRouter: jest.fn().mockReturnValue({ pathname: "/test" }),
}));

describe("Header", () => {
  it("is exported from Header", () => {
    expect(Header).toBeDefined();
  });

  it("renders as expected ", () => {
    const wrapper = shallow(<Header />);

    expect(wrapper).toMatchSnapshot();
  });
});
