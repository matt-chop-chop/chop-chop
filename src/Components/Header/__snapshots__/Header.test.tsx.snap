// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders as expected  1`] = `
<Flex
  alignItems="center"
  background="var(--chakra-colors-light-emphasis)"
  borderBottom="1px solid var(--chakra-colors-light-text)"
  position="fixed"
  px={4}
  py={2}
  width="100%"
  zIndex={1000}
>
  <Container
    maxWidth="1320px"
    width="100%"
  >
    <Flex
      alignItems="center"
    >
      <Logo />
      <Link
        _hover={
          {
            "transform": "scale(1.1)",
            "transition": "transform 250ms ease-in",
          }
        }
        as={
          {
            "$$typeof": Symbol(react.forward_ref),
            "default": [Circular],
            "render": [Function],
          }
        }
        href="/"
        ml={6}
        transition="transform 250ms ease-out"
      >
        <Text
          variant="body-large"
        >
          Home
        </Text>
      </Link>
    </Flex>
  </Container>
</Flex>
`;
